src/main/cpp/sim/Simulator.h

```diff
@@ -55,6 +55,14 @@ class Simulator
 	/// Gets the simulator's configuration.
 	SingleSimulationConfig GetConfiguration() const { return m_config; }
 
+	/// Gets the simulator's date
+	unsigned int GetDate() const;
+
+	std::vector<std::vector<Cluster>> GetClusters() const
```

Comment:
Yeah, I meant "completely change `Simulator` to use that `struct` as member." If you want to be able to iterate through the contents of that `struct`, then you can store the `std::vector<Cluster>` instances in a `private` `std::array<std::vector<Cluster>, 5>` of the `struct`, expose its iterators and provide convenient getters.